core/tutorials/Unloading/Host/Program.cs

```diff
@@ -0,0 +1,83 @@
+{
+    // This is a minimal collectible (unloadable) AssemblyLoadContext
+    class HostAssemblyLoadContext : AssemblyLoadContext
+    {
+        public HostAssemblyLoadContext() : base(isCollectible: true)
+        {
+        }
+
+        // The Load method override that returns null causes all the 
+        // dependencies of the plugin Assembly to be loaded into the
+        // default AssemblyLoadContext. So only the plugin Assembly
+        // itself is loaded into the HostAssemblyLoadContext.
+        // NOTE: it is possible to load some of the dependencies into
+        // the HostAssemblyLoadContext too, but it is beyond the scope
+        // of this sample.
+        protected override Assembly Load(AssemblyName name)
+        {
+            return null;
+        }
+    }
+
+    class Program
+    {
+        // It is important to mark this method as NoInlining, otherwise the JIT could decide
+        // to inline it into the Main method. That could then prevent successful unloading
+        // of the plugin because some of the MethodInfo / Type / Plugin.Interface / HostAssemblyLoadContext
+        // instances may get lifetime extended beyond the point when the plugin is expected to be
+        // unloaded.
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void ExecuteAndUnload(string assemblyPath, out WeakReference alcWeakRef)
+        {
+            // Create the unloadable HostAssemblyLoadContext
+            var alc = new HostAssemblyLoadContext();
+
+            // Create a weak reference to the AssemblyLoadContext that will allow us to detect
+            // when the unload completes.
+            alcWeakRef = new WeakReference(alc);
+
+            // Load the plugin assembly into the HostAssemblyLoadContext. 
+            // NOTE: the assemblyPath must be an absolute path.
+            Assembly a = alc.LoadFromAssemblyPath(assemblyPath);
+
+            // Get the plugin interface by calling the PluginClass.GetInterface method via reflection.
+            Type pluginType = a.GetType("Plugin.PluginClass");
+            MethodInfo getInterface = pluginType.GetMethod("GetInterface", BindingFlags.Static | BindingFlags.Public);
+            Plugin.Interface plugin = (Plugin.Interface)getInterface.Invoke(null, null);
+
+            // Now we can call methods of the plugin using the interface
+            string result = plugin.GetMessage();
+            Plugin.Version version = plugin.GetVersion();
+
+            Console.WriteLine($"Response from the plugin: GetVersion(): {version}, GetMessage(): {result}");
+
+            // This initiates the unload of the HostAssemblyLoadContext. The actual unloading doesn't happen
+            // right away, GC has to kick in later to collect all the stuff.
+            alc.Unload();
+        }
+
+        static void Main(string[] args)
+        {
+            WeakReference hostAlcWeakRef;
+            ExecuteAndUnload(Path.GetFullPath("Plugin.dll"), out hostAlcWeakRef);
```

Comment:
Since this is loading "Plugin" built as regular csproj, would it make sense to use the @vitek-karas plugin loading APIs here to have a more complete sample for loading plugins?

Reply:
I am thinking we may want to have two samples - that match the main scenarios from the design doc:
- Plugin loading (this one)
- Dynamic compilation (compile a piece of code in memory with Roslyn, load it, run it and unload it)

Reply:
This makes sense. Let me update this PR by adding the usage of AssemblyDependencyResolver to resolve the plugin dependencies and load them into the context instead of loading just the plugin.dll itself into it.
I will create the Dynamic compilation sample as a separate PR later.